gpytorch/lazy/kronecker_product_added_diag_lazy_tensor.py

```diff
@@ -46,8 +46,8 @@ def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True)
         return inv_quad_term, logdet_term
 
     def _logdet(self):
-        eig_matrix, _ = self.lazy_tensor._symeig()
-        evals_plus_diag = eig_matrix.diag() + self.diag_tensor.diag()
+        evals, _ = self.lazy_tensor.symeig(eigenvectors=False)
```

Comment:
From what I understand of your `@cached` changes, will having the call with `eigenvectors=False` here cause us to do another symeig? We should avoid this, since it's a strict subset of the information we would be computing had we already done a `_solve` (e.g., we don't want to `symeig` twice for every `mll` computation.

Reply:
good call - we can check the cache for the `eigenvectors =True` case and re-use those results if present

Reply:
I guess this actually does not include the `@cached` changes, these are part of the TLT PR (hint, hint).